src/main/kotlin/org/rust/ide/folding/RustFoldingBuilder.kt

```diff
@@ -6,79 +6,139 @@ import com.intellij.lang.folding.FoldingDescriptor
 import com.intellij.openapi.editor.Document
 import com.intellij.openapi.project.DumbAware
 import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiComment
 import com.intellij.psi.PsiElement
 import org.rust.lang.core.psi.*
 import org.rust.lang.core.psi.impl.RustFile
-import org.rust.lang.core.psi.util.descendentsOfType
+import org.rust.lang.core.psi.visitors.RustRecursiveElementVisitor
 import java.util.*
 
 class RustFoldingBuilder() : FoldingBuilderEx(), DumbAware {
-    override fun getPlaceholderText(node: ASTNode): String = "{...}"
+    override fun getPlaceholderText(node: ASTNode): String {
+        when {
+            node.psi is PsiComment -> return "/* ... */"
+            else -> return "{...}"
```

Comment:
Nit: small code economy can be achieved in similar cases by pulling `return` outside the conditional expression:

```
return when {
    node.psi is PsiComment -> "/* .. */"
    else -> "{ ... }"
}
```

Also I usually prefer `if` over `when`:

```
return if (node.psi is PsiComment)  "/* ... */" else "{ ... }"
```